admin/class-yoast-notification-center.php

```diff
@@ -54,74 +57,162 @@ public static function get() {
 	}
 
 	/**
-	 * Get the notifications from cookie
+	 * Initialise global notification conditions
 	 *
-	 * @return array
+	 * Conditions that are not dependent of a class concretion should be registered here.
 	 */
-	private function get_notifications_from_transient() {
-
-		// The notifications array.
-		$notifications = array();
-
-		$transient_notifications = get_transient( self::TRANSIENT_KEY );
-
-		// Check if cookie is set.
-		if ( false !== $transient_notifications ) {
+	public static function initialize_conditions() {
+		$instance = self::get();
+
+		/**
+		 * Context dependent notifications:
+		 * - Yoast_Not_Indexable_Homepage_Condition - WPSEO_OnPage, needs option information.
+		 * - Yoast_Plugin_Conflict_Condition - Yoast_Plugin_Conflict, needs plugin+conflict information.
+		 */
+
+		/**
+		 * Action Register Notification Conditionns
+		 *
+		 * Allow to hook into the notification center conditions registration.
+		 *
+		 * @param $instance Yoast_Notification_Center Instance to register condition on.
+		 */
+		do_action( 'yoast_register_notification_conditions', $instance );
+	}
 
-			// Get json notifications from cookie.
-			$json_notifications = json_decode( $transient_notifications, true );
+	/**
+	 * Register notifications of conditions
+	 *
+	 * This has to happen after the translations have been loaded.
+	 */
+	public function register_notifications() {
+		/** @var Yoast_Notification_Condition $condition */
+		foreach ( $this->notification_conditions as $condition ) {
+			$notification = $condition->get_notification();
+			// Make sure we are working with a proper Notification.
+			if ( false === ( $notification instanceof Yoast_Notification ) ) {
```

Comment:
The interface requires that a `Yoast_Notification_Condition` returns a `Yoast_Notification` when calling `get_notification` so this check is not necessary.